File: submissions/Darktt/tree.swift
import Foundation

precedencegroup PipeRight {
    associativity: left
    higherThan: AssignmentPrecedence
}

infix operator |> : PipeRight

public
func |> <T, U>(value: T, function: (T) -> U) -> U
{
    function(value)
}

struct Seed
{
    let height: Int
}

func seed() -> (Int) -> Seed
{
    {
        Seed(height: $0)
    }
}
let leafs: Array<String> = "🍏🍎🍐🍊🍋🍓🍈🍑🥭🫑".map(String.init)

func watering() -> (Seed) -> String
{
    {
        seed in
        
        let result = (0 ..< seed.height).map {
            
            layer in
            
            let spaces = String(repeating: " ", count: seed.height - layer - 1)
            let leave: (Int) -> String = {
                
                if $0 == 0 {
                    
                    return "🌟"
                }
                
                return leafs.randomElement() ?? "🍏"
            }
            let leaves: String = (0 ..< (layer + 1)).map { _ in leave(layer) }.joined()
            
            return spaces + leaves
        }
        
        return result.joined(separator: "\n")
    }
}

func grow() -> (String) -> Void
{
    {
        plant in
        
        let count: Int = plant.split(separator: "\n").count - 1
        var result: String = plant + "\n" + String(repeating: " ", count: count) + "🟫"
        
        if count >= 3 {

            result += "\n" + String(repeating: " ", count: count - 2) + "🟫🟫🟫"
        }
        
        result += "\nMerry Christmas! 🎄"
        
        print(result)
    }
}

5 |> seed() |> watering() |> grow()

File: submissions/willy08wu07/tree.c
#include <stdio.h>

int main() {
    int height = 10; // 樹的高度

    // 畫樹葉部分
    for (int i = 0; i < height; i++) {
        // 印出左側空格
        for (int j = 0; j < height - i - 1; j++) {
            printf(" ");
        }

        // 印出樹葉與裝飾
        for (int k = 0; k < (2 * i + 1); k++) {
            if (i == 0) {
                printf("★"); // 頂端的星星
            } else if (k % 3 == 0) {
                printf("o"); // 裝飾小球
            } else {
                printf("*"); // 一般樹葉
            }
        }
        printf("\n");
    }

    // 畫樹幹部分
    for (int i = 0; i < 2; i++) {
        for (int j = 0; j < height - 3; j++) {
            printf(" ");
        }
        printf("[___]\n");
    }

    return 0;
}
